sdk/core/loader/boot.cc

```diff
@@ -435,6 +502,51 @@ namespace
 			return buildMMIO();
 		}
 
+		// Privileged compartments don't have sealed objects.
+		if constexpr (!std::is_same_v<
+		                std::remove_cvref_t<decltype(sourceCompartment)>,
+		                ImgHdr::PrivilegedCompartment>)
+		{
+			if (contains(sourceCompartment.sealedObjects, target, size))
+			{
+				auto sealingType =
+				  build<uint32_t,
+				        Root::Type::RWGlobal,
+				        PermissionSet{Permission::Load, Permission::Store}>(
+				    target);
+				// TODO: This currently places a restriction that data memory
+				// can't be in the low 64 KiB of the address space.  That may be
+				// too restrictive. If we haven't visited this sealed object
+				// yet, then we should update its first word to point to the
+				// sealing type.
+				if (*sealingType >= 0x10000)
```

Comment:
What is `0x10000`?  I was sort of expecting to see something about `FirstDynamicSoftware` here or somesuch.  (Values larger than that are certainly invalid, so that could be an offset applied to the address of the export table?)

If a dynamic approach is necessary, is there no other metadata in the export table entry that could be used to discriminate processed from unprocessed rows?

Reply:
It complicates the software to use values < 2^16, and feels unnecessary given that we'd be using so much RAM just to hold the types to hit that code path - we can't easily test something that requires such a huge firmware image.